data/src/test/java/org/apache/iceberg/data/TestPartitionStatsHandler.java

```diff
@@ -591,7 +591,7 @@ private static boolean isEqual(
         && stats1.positionDeleteFileCount() == stats2.positionDeleteFileCount()
         && stats1.equalityDeleteRecordCount() == stats2.equalityDeleteRecordCount()
         && stats1.equalityDeleteFileCount() == stats2.equalityDeleteFileCount()
-        && stats1.totalRecordCount() == stats2.totalRecordCount()
+        && Objects.equals(stats1.totalRecords(), stats2.totalRecords())
```

Comment:
not directly related to this PR, but why can't we handle equals/hashcode in `PartitionStats` directly?

Reply:
`PartitionStats` has `StructLike` member for `partition`, it needs a comparator for equals(). But storing comparator for each  `PartitionStats` is not efficient. Plus we also want to keep the members of this class same as partition stats spec.

It was discussed in one of the old PR.